src/coreclr/src/jit/optimizer.cpp

```diff
@@ -3647,6 +3635,29 @@ void Compiler::optUnrollLoops()
             continue;
         }
 
+#ifdef DEBUG
+        if (compStressCompile(STRESS_UNROLL_LOOPS, 50))
+        {
+            // In stress mode, quadruple the size limit, and drop
+            // the restriction that loop limit must be vector element count.
+
+            unrollLimitSz *= 4;
+        }
+        else
+#endif
+            // Unroll only if limit is 0, 1, or Vector_.Length (as a heuristic, not for correctness/structural reasons)
+            if (!(loopFlags & LPFLG_SIMD_LIMIT))
+        {
+            if (totalIter <= 1)
+            {
+                unrollLimitSz *= 4;
+            }
+            else
+            {
+                continue;
+            }
+        }
```

Comment:
Why do we increase the unroll limit in this case? Shouldn't we only do that in the stress mode?

Shouldn't this just be:
```
    // Unroll only if limit is 0, 1, or Vector_.Length (as a heuristic, not for correctness/structural reasons)
    if (!(loopFlags & LPFLG_SIMD_LIMIT) && (totalIter > 1))
    {
        continue;
    }
```

Reply:
Could you please re-review the recent changes?